sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,206 @@
+        if cat_features is None:
+            return
+
+        cat_features = check_array(cat_features, ensure_2d=False, dtype=int)
+        cat_features = column_or_1d(cat_features)
+
+        if len(set(cat_features)) != cat_features.shape[0]:
+            raise ValueError("Duplicate categorical column indices found.")
+
+        if np.all(cat_features >= 0) and np.all(cat_features < n_features):
+            return
+
+        raise ValueError("Invalid categorical feature index.")
+
+    def transform(self, X, y=None):
+        """Discretizes the data, ignoring any categorical features.
+
+        Parameters
+        ----------
+        X : numeric array-like, shape (n_samples, n_features)
+            Data containing features to be discretized.
+
+        Returns
+        -------
+        T : float array-like, shape (n_samples, n_features)
+            Categorical features will not be transformed.
+        """
+        check_is_fitted(self, ["mins_", "maxes_", "n_features_"])
+        X = check_array(X, dtype=float, ensure_2d=False)
+
+        if X.ndim == 1:
+            raise ValueError("Reshape your data.")
+
+        if X.shape[1] != self.n_features_:
+            raise ValueError("Transformed array does not have correct number "
+                             "of features. Expecting {}, received {}."
+                             .format(self.n_features_, X.shape[1]))
+
+        if self.categorical_features:
+            cat_features = np.asarray(sorted(self.categorical_features))
+        else:
+            cat_features = np.array([], dtype='int64')
+
+        # Manipulate the input attributes so categorical features can be
+        # ignored in transformation process. Also avoid divide by zero errors.
+        same_min_max = np.arange(self.n_features_)[self.maxes_ == self.mins_]
+        if len(same_min_max) != 0:
+            import warnings
+            warnings.warn("Fitted X contained continuous features at indices "
+                          "{}. These features will be discretized to the "
+                          "same value."
+                          .format(", ".join(str(i) for i in same_min_max)))
+
+        ignored_features = np.concatenate((cat_features, same_min_max))
+
+        mins = self.mins_.copy()
+        mins[ignored_features] = 0
+
+        maxes = self.maxes_.copy()
+        maxes[ignored_features] = 1
+
+        if isinstance(self.n_bins, numbers.Number):
+            n_bins = np.ones(self.n_features_) * self.n_bins
+        else:
+            n_bins = check_array(self.n_bins, dtype=int, ensure_2d=False,
+                                 copy=True)
+        n_bins[ignored_features] = 1
+
+        X_t = (X - mins) * n_bins // (maxes - mins)
+
+        # Need to make sure outliers are within boundary, but also make sure
+        # categorical features can be ignored.
+        numeric_features = np.delete(np.arange(self.n_features_), cat_features)
+        clip_min = np.repeat(-np.inf, self.n_features_)
```

Comment:
Assuming you're referring to [this comment](https://github.com/scikit-learn/scikit-learn/pull/7668/files#r101936287), we can do `X_t = X_t.astype(int)` on line 203, after clipping.